eclipse-collections/src/main/java/org/eclipse/collections/impl/set/immutable/AbstractImmutableSet.java

```diff
@@ -456,6 +458,11 @@ public boolean isProperSubsetOf(SetIterable<? extends T> candidateSuperset)
     @Override
     public ParallelUnsortedSetIterable<T> asParallel(ExecutorService executorService, int batchSize)
     {
-        return this.toSet().asParallel(executorService, batchSize);
+        Objects.requireNonNull(executorService);
+        if (batchSize < 1)
+        {
+            throw new IllegalArgumentException("batchSize must be greater than zero, but was: " + batchSize);
+        }
+        return new NonParallelUnsortedSetIterable<>(this);
```

Comment:
Is this overridden? There a possible efficient implementation for ImmutableUnifiedSet.

Reply:
Yes indeed, there are two overrides for this in `ImmutableUnifiedSet` and `ImmutableUnifiedSetWithHashingStrategy` (delegating to the wrapped `UnfiedSet`).

I haven't looked into other subclasses, the purpose here was really to avoid the costly call to `toSet()`.